decidim-proposals/app/services/decidim/proposals/proposal_search.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+module Decidim
+  module Proposals
+    # A service to encapsualte all the logic when searching and filtering
+    # proposals in a participatory process.
+    class ProposalSearch
```

Comment:
Meh, ok. As this filter grows more complex, I'm sure we'll end up composing it by several queries though.